src/styles/main.scss

```diff
@@ -1,3 +1,70 @@
+@use "../styles/utils/variables.scss" as *;
+
 body {
   margin: 0;
+  width: 100%;
+  height: 100vh;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+}
+
+.calendar {
+  display: flex;
+  flex-direction: row;
+  flex-wrap: wrap;
+  gap: $gap-size $gap-size;
+  width: $cell-size * $days-in-week + $gap-size*($days-in-week - 1);
+
+  @for $index from 1 through $max-month-length {
+    &__day:nth-child(#{$index}) {
+      box-sizing: border-box;
+      height: $cell-size;
+      width: $cell-size;
+      background-color: $base-color;
+      border: 1px solid #000;
+      display: flex;
+      justify-content: center;
+      align-items: center;
+      transition: transform 0.5s;
+      cursor: pointer;
+    }
+  }
+
+  @for $index from 1 through $max-month-length {
+    &__day:nth-child(#{$index})::after {
+      content: "#{$index}";
+      font-family: Arial, Helvetica, sans-serif;
+      font-size: 30px;
+    }
+  }
+
+  &__day:hover {
+    background-color: $hover-color;
+    transform: translate(0, -20px);
+  }
+
+  @for $index from 28 through 31 {
+    &--#{$index} {
+      @for $newindex from ($index+1) through 31 {
+        .calendar__day:nth-child(#{$newindex}) {
+          display: none;
+        }
+      }
+    }
+  }
+
+  &--31 {
+    .calendar__day:nth-child(31) {
+      display: flex;
+    }
+  }
```

Comment:
make it work in the loop
without this 'fix'
![image](https://user-images.githubusercontent.com/99331793/188105339-21b0e950-9d16-4bbd-a07b-e224df3c326a.png)